src/crags/services/comments.service.ts

```diff
@@ -96,9 +98,36 @@ export class CommentsService {
       options.where['type'] = params.type;
     }
 
+    if (!showHiddenCrags) {
+      options.relations = {
+        crag: true,
+      };
+      options.where['crag'] = {
+        isHidden: false,
+      };
+    }
+
     options.order = { created: 'DESC' };
 
-    return this.commentsRepository.find(options);
+    let count: number;
+    if (params.pageSize != null && params.pageNumber != null) {
+      count = await this.commentsRepository.count(options);
+      options.take = params.pageSize;
+      options.skip = params.pageSize * (params.pageNumber - 1);
+    }
+
```

Comment:
This is debatable, but for now I'd rather keep it this way because this is the shape of the pagination object that we use elsewhere and it's the shape of the pagination that is used on the client.

Reply:
done, used logic from difficulty votes